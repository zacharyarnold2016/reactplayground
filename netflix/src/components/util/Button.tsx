import React from "react";
import { button } from "../../interfaces/Button.interface";

const Button = (props: button) => {
  return (
    <button
      type={props.purpose}
      className={props.style}
      onClick={props.callback || null}
    >
      {props.text}
    </button>
  );
};

export default Button;
