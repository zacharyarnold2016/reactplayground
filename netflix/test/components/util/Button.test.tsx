import renderer from "react-test-renderer";
import Button from "../../../src/components/util/Button";
import { expect } from "@jest/globals";
import {
  ButtonStyle,
  ButtonType,
} from "../../../src/interfaces/components/util/Button.interface";

describe("Button", () => {
  const testCallback = jest.fn();
  test("Button Clicked", () => {
    const component = renderer.create(
      <Button
        type={ButtonType.BUTTON}
        styling={ButtonStyle.A}
        callback={testCallback}
      />
    );
    let tree: any = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.onClick();
    expect(testCallback).toBeCalled();
  });
  test("No Callback", () => {
    const component = renderer.create(
      <Button type={ButtonType.BUTTON} styling={ButtonStyle.A} />
    );
    let tree: any = component.toJSON();
    expect(tree.props.onClick).toBe(null);
  });
});
