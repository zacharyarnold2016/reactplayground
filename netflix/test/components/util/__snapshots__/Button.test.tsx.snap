// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button Clicked 1`] = `
<button
  className="primaryButton"
  onClick={[MockFunction]}
/>
`;
